### DISCUSSION TITLE
Replace kobold lite?

### DISCUSSION THREAD

**Discussion by aolko** [LostRuins/koboldcpp#953] [Q&A]
Kobold lite has a bad UI/UX. How does one replace it or modify it?

**Comment by LostRuins** [comment]
First, have you tried using the other UI modes in Kobold Lite, such as the aesthetic mode? You can set those in the settings. 

![image](https://github.com/LostRuins/koboldcpp/assets/39025047/462c4ed2-f09e-4bca-8999-62a4c4483191)

If you still want to modify it, you can edit the klite.embd file, which is the single HTML file that contains Kobold Lite.

If you want a third party ui, you can try downloading SillyTavern, LibreChat or OpenWebUI, and then connect it to KoboldCpp via the provided endpoint url you see in the KoboldCpp terminal.

**Reply by aolko** [reply]
i'd like to have a mix of intuitive GPTs (see: deepgame) and agnai/novelai in the UI.
As for
> LibreChat or OpenWebUI

they are only designed for conversations, not story writing

**Reply by LostRuins** [reply]
Maybe you can elaborate what you don't like about Kobold Lite right now. The UI is actually quite customizable, did you try the aesthetic mode? 

![image](https://github.com/LostRuins/koboldcpp/assets/39025047/89b6b2ca-eede-4efd-b3fb-ff04c5bca6ff)

**Reply by aolko** [reply]
> Maybe you can elaborate what you don't like about Kobold Lite right now. The UI is actually quite customizable, did you try the aesthetic mode?
> 
> ![image](https://private-user-images.githubusercontent.com/39025047/344258362-89b6b2ca-eede-4efd-b3fb-ff04c5bca6ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk2MTg1NjQsIm5iZiI6MTcxOTYxODI2NCwicGF0aCI6Ii8zOTAyNTA0Ny8zNDQyNTgzNjItODliNmIyY2EtZWVkZS00ZWZkLWIzZmItZmYwNGM1YmNhNmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDIzNDQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjN2JiZjBkOTU3OWMyNmQzYzhiNjNhYzcyMDhhYTM1NGQ4YWVmOTMwN2FlODI2ZTFlZDdiYzg5ZDhkYzcwMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.r6uzKTC4AZ1Zz_wKOGd11CSdSzhxhodHBV-rj20P4Z8)

yep, and it's janky, it's like [level 2.5-ish](https://youtu.be/JZHH0Yt5V_g) designer did it

**Reply by aolko** [reply]
Here are the bullet points why KoboldAI Lite is jank:

- Cluttered layout: The interface contains many buttons, sliders, and text fields crammed into a relatively small space, making it feel cluttered and overwhelming.
- Inconsistent styling: There's a mix of different button styles, colors, and text sizes that don't create a cohesive visual design.
- Limited color palette: The interface primarily uses shades of blue and gray, which can make it feel monotonous and lacking visual interest.
- Poor contrast: Some text and elements have low contrast against the background, potentially impacting readability.
- Unclear hierarchy: It's not immediately obvious what the most important elements or actions are, as everything seems to have equal visual weight.
- Small text and controls: Some text and UI elements appear quite small, which could make them difficult to read or interact with.
- Lack of whitespace: There's very little breathing room between elements, contributing to the cluttered feel.
- Dated appearance: The overall aesthetic feels somewhat outdated, reminiscent of older web interfaces rather than modern, sleek designs.
- Modal windows: The use of modal windows for settings and other functions can interrupt the user's workflow and feels clunky.
- Unclear labeling: Some buttons and fields have unclear or technical labels that may not be immediately understandable to all users.

<details><summary>Bullet points about Agnaistic</summary>
<p>

- Dark mode: The interface uses a dark color scheme, which can be easier on the eyes and is popular among many users.
- Sidebar navigation: There's a consistent sidebar on the left for easy navigation between different sections like Characters, Chats, Library, etc.
- Chat interface: The main area displays a chat interface, allowing users to interact with AI characters.
- Character customization: Image 2 shows a character editing interface, allowing users to customize AI personas with names, descriptions, and avatars.
- Memory features: Image 1 shows options for creating "Memory Books" and embedding articles, which could enhance the context and knowledge of AI characters.
- Responsive design: The layout appears to adapt well to different content, with collapsible sections and modal windows.
- Clear typography: The text is legible and well-structured, with appropriate hierarchy.
- Minimalist design: The interface is relatively clean and uncluttered, focusing on core functionality.
- Customization options: Users seem to have control over various aspects of their experience, from character creation to memory management.
- Innovative features: The combination of AI chat, character creation, and memory management suggests a unique and potentially engaging user experience.

![ViewGrabber - Agnaistic AI - A Character AI Alternative With No Filters Janitor AI Alternative  vsW028xIRGU - 1397x786 - 4m51s](https://github.com/LostRuins/koboldcpp/assets/581458/8a3ab1db-8326-463d-a954-92e240334025)
![AI Chat Bots Are Cool - Agnaistic 10-minute quick start  JgT2jLMSOP0 - 1397x786 - 6m33s](https://github.com/LostRuins/koboldcpp/assets/581458/9fa98287-d44d-4638-b212-7e7cfbe09e70)

</p>
</details> 

<details><summary>And about NovelAI</summary>
<p>

- Dark theme: The dark color scheme is easy on the eyes and creates an immersive atmosphere.
- Clear layout: The main story area is prominent in the center, with tools and options on the right side.
- Contextual options: The right panel shows relevant controls like "Context," "Phrase Bias," and "Banned Tokens" for fine-tuning the story generation.
- Interactive elements: The bottom of the screen has a dialog box for user input, with options to "DO" or "SAY" things in the story.
- Customization: There are many options for tailoring the AI's output, such as adjusting the AI model, modules, and story settings.
- Progress tracking: The interface shows the user's inventory and actions taken, helping to keep track of the story progression.
- Minimalist design: The UI is clean and uncluttered, focusing on the story content.
- Accessibility: Tools like "Lorebook Quick Access" suggest ease of use for managing complex narratives.

![P Gatcomb - NovelAI - Text Adventure Tips and Shortcuts  UYZAPnsfUkg - 1397x786 - 6m13s](https://github.com/LostRuins/koboldcpp/assets/581458/66a55865-eea7-4722-b0b9-aca7a6f62d22)
![P Gatcomb - NovelAI - Text Adventure Tips and Shortcuts  UYZAPnsfUkg - 1397x786 - 10m15s](https://github.com/LostRuins/koboldcpp/assets/581458/2bfeb512-f7d6-45ce-ac07-925758896acf)


</p>
</details>

**Reply by LostRuins** [reply]
Speaking of Agnaistic, you can actually connect that to koboldcpp too, as the endpoint is supported. 

I don't know if I agree with all the points you raised, tbh that list read to me like it was AI generated and not an actual critique. 

Kobold Lite is cluttered, true, but uses a consistent font and color scheme by default, unless you choose to customize it. But in any case, I am not a designer, so if you'd like to contribute, PRs are always welcome.

**Reply by aolko** [reply]
> Speaking of Agnaistic, you can actually connect that to koboldcpp too, as the endpoint is supported.

True, but then again, focused on conversations, not story writing. Although it allows having multiple bots per chat, you have to resort to ugly hacks like "narrator bot".
